dev/start.fish

```diff
@@ -0,0 +1,385 @@
+    set -e _MACH
+    set -e _NAME
+    set -e _PYTHON
+    set -e _PYTHONEXE
+    set -e _SCRIPT_PATH
+    set -e _SHELL_TYPE
+    set -e _SRC
+    set -e _UPDATE
+    set -e _UPDATED
+end
+
+function updating
+    # check if explicitly updating or if 24 hrs since last update
+    test "$_UPDATE" = "true" && return 0
+    test -f "$_UPDATED" || return 0
+    set current_time (date +%s)
+    set file_time (date -r "$_UPDATED" +%s)
+    set diff_time (math $current_time - $file_time)
+    test $diff_time -lt 86400
+end
+
+cleanup
+
+# get source path
+set _SCRIPT_PATH (status filename)
+set _SRC (dirname (dirname $_SCRIPT_PATH))
+set _SRC (cd "$_SRC" 2>&1 > /dev/null; pwd -P)
+
+# define default values
+set _DEFAULT_PYTHON "3.10"
+set _DEFAULT_INSTALLER "miniconda"
+# $ uname -sm
+# Linux x86_64
+# Linux aarch64
+# Darwin arm64
+# ...
+set _DEFAULT_MACH (uname)
+set _DEFAULT_ARCH (uname -m)
+set _DEFAULT_UPDATE "false"
+set _DEFAULT_DEVENV "$_SRC/devenv"
+set _DEFAULT_DRYRUN "false"
+
+# parse args
+set i 1
+while test $i -le (count $argv)
+    switch $argv[$i]
+        case -p --python
+            set _PYTHON $argv[(math $i + 1)]
+            set i (math $i + 2)
+        case -m --mach
+            set _MACH $argv[(math $i + 1)]
+            set i (math $i + 2)
+        case -a --arch
+            set _ARCH $argv[(math $i + 1)]
+            set i (math $i + 2)
+        case -i --installer
+            set _INSTALLER_TYPE $argv[(math $i + 1)]
+            set i (math $i + 2)
+        case -u --update
+            set _UPDATE "true"
+            set i (math $i + 1)
+        case -d --devenv
+            set _DEVENV $argv[(math $i + 1)]
+            set i (math $i + 2)
+        case -n --dry-run
+            set _DRYRUN "true"
+            set i (math $i + 1)
+        case -h --help
+            echo "Usage: source $_SCRIPT_PATH [options]"
+            echo ""
+            echo "Options:"
+            echo "  -p, --python    VERSION    Python version for the env to activate. (default: $_DEFAULT_PYTHON)"
+            echo "  -i, --installer INSTALLER  Installer to use: miniconda or miniforge, can also be defined in ~/.condarc. (default: $_DEFAULT_INSTALLER)"
+            echo "  -m, --mach      MACHINE    Machine type for the env to activate. (default: $_DEFAULT_MACH)"
+            echo "  -a, --arch      ARCH       Architecture for the env to activate. (default: $_DEFAULT_ARCH)"
+            echo "  -u, --update               Force update packages. (default: $_DEFAULT_UPDATE, update every 24 hours)"
+            echo "  -d, --devenv    PATH       Path to base env install, can also be defined in ~/.condarc."
+            echo "                             Path is appended with machine hardware name, see --mach. (default: $_DEFAULT_DEVENV)"
+            echo "  -n, --dry-run              Display env to activate. (default: $_DEFAULT_DRYRUN)"
+            echo "  -h, --help                 Display this."
+            return 0
```

Comment:
I get an error when I source this script in a fish shell `3.x`. It looks like maybe this should be [`exit 0`](https://fishshell.com/docs/current/cmds/exit.html) instead. This is also compatible with fish 4.x. I think this change needs to happen for every `return` statement.

```suggestion
            exit 0
```